src/helpers/QInputSelectionHelper.ts

```diff
@@ -0,0 +1,72 @@
+import { QInput } from "quasar";
+import { Ref } from "vue";
+
+/**
+ * QInput の選択範囲への操作を簡単にできるようにするクラス
+ */
+export class QInputSelectionHelper {
+  private _nativeEl: HTMLInputElement | undefined = undefined;
+
+  constructor(private textfield: Ref<QInput | undefined>) {}
+
+  // this.start が number | null なので null も受け付ける
+  setCursorPosition(index: number | null) {
+    if (index === null) return;
+
+    this.nativeEl.selectionStart = this.nativeEl.selectionEnd = index;
+  }
+
+  getReplacedStringTo(string: string, allowInsertOnly = false) {
+    if (!allowInsertOnly && this.isEmpty) {
+      return this.nativeEl.value;
+    }
+
+    return `${this.startFragment}${string}${this.endFragment}`;
+  }
```

Comment:
`string: string`はちょっと衝突しそうで危険かもです。`str`にするとか？
```suggestion
  getReplacedStringTo(str: string, allowInsertOnly = false) {
    if (!allowInsertOnly && this.isEmpty) {
      return this.nativeEl.value;
    }

    return `${this.startFragment}${str}${this.endFragment}`;
  }
```